pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -138,6 +140,24 @@ func validateStepOverrides(overrides []TaskRunStepOverride) (errs *apis.FieldErr
 	return errs
 }
 
+// validateStepOverridesResourceRequirements validates if both step-level and task-level resource requirements are configured
+func validateStepOverridesResourceRequirements(computeResources corev1.ResourceRequirements, overrides []TaskRunStepOverride) (errs *apis.FieldError) {
+	for _, override := range overrides {
+		if override.Resources.Size() > 0 && isResourceRequirementsInTaskRun(computeResources) {
+			return &apis.FieldError{
+				Message: "TaskRun can't be configured with both step-level (stepOverrides.resources) and task-level (spec.computeResources) resource requirements",
+				Paths:   []string{"resources"},
+			}
+		}
+	}
+	return nil
+}
+
+// isResourceRequirementsInTaskRun checks if resource requirements are configured in TaskRun under task-level
+func isResourceRequirementsInTaskRun(computeResources corev1.ResourceRequirements) bool {
```

Comment:
this function is not needed; just check if taskrun.computeResources is a nil pointer

Reply:
right. this will apply nicely after updating as a pointer.